### PR TITLE
Skip asserts in Http SSL test on macOS

### PR DESCRIPTION
**Pull request by stephentoub**
Closes https://github.com/dotnet/corefx/issues/21922
cc: @bartonjs

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by stephentoub** [comment]
@dotnet/dnceng, Jenkins doesn't appear to be responding to re-test requests.  Something special I need to do?

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by mmitche** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs [hunk 1]
```diff
@@ -157,17 +158,25 @@ public static IEnumerable<object[]> UseCallback_ValidCertificate_ExpectedValuesD
...
                     X509ChainStatusFlags flags = chain.ChainStatus.Aggregate(X509ChainStatusFlags.NoError, (cur, status) => cur | status.Status);
-                    Assert.True(errors == SslPolicyErrors.None, $"Expected {SslPolicyErrors.None}, got {errors} with chain status {flags}");
+                    bool skipAssertsForMacOS = // https://github.com/dotnet/corefx/issues/21922#issuecomment-315555237
+                        RuntimeInformation.IsOSPlatform(OSPlatform.OSX) &&
+                        checkRevocation &&
+                        errors == SslPolicyErrors.RemoteCertificateChainErrors &&
+                        flags == X509ChainStatusFlags.RevocationStatusUnknown;
+                    if (!skipAssertsForMacOS)
+                    {
+                        Assert.True(errors == SslPolicyErrors.None, $"Expected {SslPolicyErrors.None}, got {errors} with chain status {flags}");
```
Comment (bartonjs) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: This assert should be the only one you need to skip.  The rest should hold.

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Release Build

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: Changed.

**Comment by stephentoub** [comment]
@dotnet/dnceng, a lot of OSX legs are failing on different PRs with errors like:
```
Error during cleanup: java.io.IOException: remote file operation failed: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest at hudson.remoting.Channel@19413640:JNLP4-connect connection from 131.107.19.218/131.107.19.218:50077: hudson.remoting.ChannelClosedException: channel is already closed
```

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build please
@dotnet-bot test OSX x64 Debug Build please

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop OSX x64 Debug Build please
@dotnet-bot test OSX x64 Debug Build please
@dotnet-bot test Outerloop OSX x64 Release Build

**Comment by stephentoub** [comment]
@dotnet/dnceng, @mmitche, the OSX legs are still failing even after Matt reset the boxes, e.g. "Using context: OSX x64 Debug Build
java.nio.channels.ClosedChannelException"

**Comment by mmitche** [comment]
@dotnet-bot test OSX x64 Debug Build please

**Comment by mmitche** [comment]
@stephentoub I think it actually might just be one of the machines.

**Comment by stephentoub** [comment]
> I think it actually might just be one of the machines.

Ok, thanks.  Can we take it out of the rotation until it's fixed?

**Comment by mmitche** [comment]
@stephentoub Done.  Can you let me know whether you see it again.  @dotnet/dnceng for visibility

**Comment by stephentoub** [comment]
> Can you let me know whether you see it again

Will do. Thanks.

**Pull request merged**